source/GlassView.Export/DirectoryExport.cs

```diff
@@ -1,21 +1,22 @@
 using System.Text.Json;
 using BenchmarkDotNet.Loggers;
+using Atmoos.World;
 using Atmoos.GlassView.Core.Models;
 
 namespace Atmoos.GlassView.Export;
 
-internal sealed class DirectoryExport(DirectoryInfo path, JsonSerializerOptions options) : IExport
+internal sealed class DirectoryExport<FileSystem>(IDirectory directory, JsonSerializerOptions options) : IExport
+    where FileSystem : IFileCreation
 {
     public async Task Export(BenchmarkSummary summary, ILogger logger, CancellationToken token)
     {
-        FileInfo file = path.AddFile(FileNameFor(summary));
-        logger.WriteLineInfo($"file: {file.FullName}");
-        using var stream = file.Open(FileMode.Create, FileAccess.Write, FileShare.None);
+        IFile file = FileSystem.Create(directory, FileNameFor(summary));
```

Comment:
I'm not sure about the types. I'd have to look into what IFile contains and what FileInfo contains.
Personally, I'm inclined to introduce as few "own" types as possible, so that, if required, we could easily switch them. It also means that we have less code to maintain and that we get the benefits of the type improvements, if the type evolves over time.

Reply:
What would be the advantages of IFile relative to FileInfo (besides the advantages that an interface would bring, aka it can be mocked)?

Reply:
I just discovered that IFile comes from Microsoft. Nevermind the questions 😅

Reply:
IFile is a type I defined in Atmoos.World: https://github.com/atmoos/World/blob/main/source/Atmoos.World/IFile.cs

It's a minimal abstraction over a file, that supports reading a limited amount of metadata and, natuarally reading & writing, but not deleting.

Reply:
FileInfo is a type that I've come to consider absolutely useless, because of an utterly confusing API:

```csharp
var info = new FileInfo("some/path/to/new/file.txt");
_ = info.Create();
Console.WriteLine(info.Exists); // false
```
or conversely:

```csharp
var info = new FileInfo("some/path/to/existing/file.txt");
info.Delete();
Console.WriteLine(info.Exists); // true
```

This is due to the fact that it does not refresh itself, one always needs to call `info.Refresh()` to get reliable results, that is defined in a parent class [here](https://learn.microsoft.com/en-us/dotnet/api/system.io.filesysteminfo.refresh)...
Hence in Atmoos.World I do that every time I access the actual file system: https://github.com/atmoos/World/blob/main/source/Atmoos.World.IO/FileSystem/File.cs